src/components/Page.tsx

```diff
@@ -1,8 +1,37 @@
-import React, { Component } from "react";
+import PropTypes, { ValidationMap } from "prop-types";
+import React, { Attributes, ComponentType } from "react";
+
+export interface PageComponent<P = {}> {
+  id: string;
+  Component: string | ComponentType<Attributes | P>;
+  props?: P;
+}
+
+export interface PageProps<P = {}> {
+  components: PageComponent<P>[];
+}
+
+export class Page<P> extends React.Component<PageProps<P>> {
+  static propTypes: ValidationMap<PageProps> = {
```

Comment:
@CJWaszczuk See this for how we can assert prop types on the `static propTypes`.